packages/fluentui/react-northstar/src/components/Menu/Menu.tsx

```diff
@@ -110,11 +110,9 @@ export interface MenuProps extends UIComponentProps, ChildrenComponentProps {
 
 export const menuClassName = 'ui-menu';
 
-export type MenuStylesProps = Required<
-  Pick<
-    MenuProps,
-    'iconOnly' | 'fluid' | 'pointing' | 'pills' | 'primary' | 'underlined' | 'vertical' | 'submenu' | 'secondary'
-  >
+export type MenuStylesProps = Pick<
+  MenuProps,
+  'iconOnly' | 'fluid' | 'pointing' | 'pills' | 'primary' | 'underlined' | 'vertical' | 'submenu' | 'secondary'
 >;
 
 function useActualOnItemClick<P>(onItemClick: P) {
```

Comment:
@layershifter @assuncaocharles  shouldn't we call here `useLayoutEffect`?
```
function useActualOnItemClick<P>(onItemClick: P) {
  const actualOnItemClick = React.useRef<P>(onItemClick);
  React.useEffect(() => { // useLaytoutEffect?
    actualOnItemClick.current = onItemClick;
  });
  return actualOnItemClick;
}
```
We can ensure that way that it will be invoked before any other `useEffects`​​​​​​​ we may have or add in the future

Reply:
It makes sense since it can be dep for `useEffect`. I guess we will have to use `useIsomorphicLayoutEffect` instead `useLayoutEffect`?

Reply:
@mnajdova `useLayoutEffect` can't be used with SSR and only `useEffect` can be used there. It means that the behavior will be different between client and server. That's why I proposed to use `useEffect`.